packages/web/lib/browser.js

```diff
@@ -8,20 +10,63 @@ const {
   resolveFn,
   resolveFnScope,
   resolveInternalScope,
+  validateScope,
 } = require("@browselang/core/lib/scope");
 const { help } = require("@browselang/core/lib/utils");
 const UrlPattern = require("url-pattern");
 const { BrowseError } = require("@browselang/core/lib/error");
 
 const getPageScope = require("./page");
 
+const throws = (fn) => (...args) => {
+  try {
+    return { success: true, value: fn(...args) };
+  } catch (e) {
+    return { success: false };
+  }
+};
+
+const strictValidateScope = (scope, message) => {
+  if (!validateScope((scope) => scope.internal.isPage, scope)) {
```

Comment:
also, it seems this should really be `assertInPageScope`, since you've hard coded the predicate